common.mk

```diff
@@ -0,0 +1,70 @@
+# Common definitions
+
+BIN := $(GDK)/bin
+LIB := $(GDK)/lib
+
+SRC_LIB := $(GDK)/src
+RES_LIB := $(GDK)/res
+INCLUDE_LIB := $(GDK)/inc
+MAKEFILE_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
+MAKEFILE_DIR := $(subst \,/,$(MAKEFILE_DIR))
+
+ifeq ($(OS),Windows_NT)
+	# Native Windows
+	SHELL := $(BIN)/sh.exe
+	RM := $(BIN)/rm.exe
+	CP := $(BIN)/cp.exe
+	MKDIR := $(BIN)/mkdir.exe
+
+	AR := $(BIN)/ar.exe
+	CC := $(BIN)/gcc.exe
+	LD:= $(BIN)/ld.exe
+	NM:= $(BIN)/nm.exe
+	OBJCPY := $(BIN)/objcopy.exe
+	ASMZ80 := $(BIN)/sjasm.exe
+	MACCER := $(BIN)/mac68k.exe
+	BINTOS := $(BIN)/bintos.exe
+	LTO_PLUGIN := --plugin=liblto_plugin-0.dll 
+	LIBGCC := $(LIB)/libgcc.a
+else
+	ifeq ($(SGDK_DOCKER),y)
+		# Linux docker
+		SHELL = sh
+		RM = rm
+		CP = cp
+		MKDIR = mkdir
+
+		AR := $(BIN)/ar
+		CC := $(BIN)/gcc
+		LD := $(BIN)/ld
+		NM := $(BIN)/nm
+		OBJCPY := $(BIN)/objcopy
+		ASMZ80 := $(BIN)/sjasm
+		MACCER := $(BIN)/mac68k
+		BINTOS := $(BIN)/bintos
+		LTO_PLUGIN := --plugin=liblto_plugin-0.dll
+		LIBGCC := $(LIB)/libgcc.a
```

Comment:
the docker setup could also use native linux binaries instead of wine

Reply:
Yes, in fact I would like that to be the case, and I have an "experimental" Docker that does not use Wine at all. I would like to test it a bit more, and before sending the PR I need this to get merged, because it does not work without this change, and that is the reason that the Dockerfile currently downloads SGDK from my fork instead of downloading it from upstream. Note that the Dockerfile is using GCC 11.2 and OpenJDK 17.

You can view the Dockerfile here: https://gitlab.com/doragasu/docker-sgdk
And if you want to give it a try, you can pull it directly from GitLab registry as documented in the README.md.

Reply:
@Stephane-D can you create an account in https://hub.docker.com/ ?
So it can be used as the official docker repository for SGDK. 
After that, if you add the token to the github secrets, we cand send a PR to this repository adding a github action that builds the SGDK docker and pushes it to the repository on every PR.
An additional action (manually triggered) can be added for stable library releases.
WDYT?

Reply:
@Stephane-D @pladaria an official docker image released on https://hub.docker.com/ would be awesome it would be very handy to setup a ci/cd systems for ones megadrive games (e.g.: using github actions).
@Stephane-D do you think that would be feasable for you?

Reply:
> @Stephane-D @pladaria an official docker image released on https://hub.docker.com/ would be awesome. It would be very handy to setup a ci/cd systems for ones megadrive games (e.g.: using github actions). @Stephane-D do you think that would be feasable for you?

You can have a look to my SGDK docker: https://gitlab.com/doragasu/docker-sgdk/
It is uploaded to GitLab registry (so you do not need to build it) and I use it with GitLab-CI to automatically build ROMs. It is not official though.